caas/kubernetes/provider/application/constraints.go

```diff
@@ -266,6 +273,165 @@ func processPodAffinity(pod *core.PodSpec, affinityLabels map[string]string) err
 	return nil
 }
 
+const (
+	topologySpreadMaxSkew         = "maxSkew"
+	topologySpreadMinDomains      = "minDomains"
+	topologySpreadNodeTaintPolicy = "nodeTaintsPolicy"
+	topologySpreadMatchLabels     = "matchLabelKeys"
+)
+
+func processTopologySpreadConstraints(pod *core.PodSpec, affinityLabels map[string]string, appName string) error {
+	topologySpreadValues := make(map[string]string)
+
+	validKeys := set.NewStrings(
+		topologySpreadMaxSkew,
+		topologySpreadMinDomains,
+		topologySpreadNodeTaintPolicy,
+		topologySpreadMatchLabels,
+		topologyKeyTag,
+	)
+
+	for key, value := range affinityLabels {
+		if !strings.HasPrefix(key, topologySpreadPrefix) {
+			continue
+		}
+
+		key = strings.TrimPrefix(key, topologySpreadPrefix)
+		if !validKeys.Contains(key) {
+			return errors.Errorf("invalid topology spread constraint key %q", key)
+		}
+		if key == topologySpreadMaxSkew {
+			val, err := strconv.Atoi(value)
+			if err != nil {
+				return errors.Errorf("invalid value %q for topology spread max skew", value)
+			}
+			if val < 1 {
+				return errors.Errorf("invalid value, maxSkew must be greater or equal to 1")
+			}
+		}
+		topologySpreadValues[key] = value
+	}
+	if len(topologySpreadValues) == 0 {
+		return nil
+	}
+	_, present := topologySpreadValues[topologyKeyTag]
+	if !present {
+		return errors.Errorf("topology-key not set for topology spread constraints: %v", affinityLabels)
+	}
+
+	var appNameLabel string = "app.kubernetes.io/name"
+
+	updateTopologyTerm := func(topologyTerms *core.TopologySpreadConstraint, tags map[string]string) {
+		// Sort for stable ordering.
+		var keys []string
+		for k := range tags {
+			keys = append(keys, k)
+		}
+		sort.Strings(keys)
+		var (
+			labelSelector    v1.LabelSelector
+			topologyKey      string = "kubernetes.io/zone"
+			maxSkew          int    = 1
+			minDomains       int    = 3
+			nodeTaintsPolicy *core.NodeInclusionPolicy
+		)
+
+		for _, key := range keys {
+
+			if key == topologyKeyTag {
```

Comment:
The structure here is nice than before, but instead of if blocks with continue, use `switch` with the key conversion to match expressions done in the default block.

Reply:
ack, will update for the switch case